src/DotNet.Testcontainers/Builders/CredsHelperProvider.cs

```diff
@@ -1,20 +1,66 @@
 ﻿namespace DotNet.Testcontainers.Builders
 {
+  using System.Text.Json;
   using DotNet.Testcontainers.Configurations;
+  using Microsoft.Extensions.Logging;
 
-  /// <inheritdoc cref="IAuthenticationProvider{TAuthenticationConfiguration}" />
-  internal sealed class CredsHelperProvider : IAuthenticationProvider<IDockerRegistryAuthenticationConfiguration>
+  /// <inheritdoc />
+  internal sealed class CredsHelperProvider : IDockerRegistryAuthenticationProvider
   {
+    private const string TokenUsername = "<token>";
+    private readonly JsonElement dockerConfig;
+    private readonly ILogger logger;
+
+    public CredsHelperProvider(JsonElement dockerConfig, ILogger logger)
+    {
+      this.dockerConfig = dockerConfig;
+      this.logger = logger;
+    }
+
     /// <inheritdoc />
-    public bool IsApplicable()
+    public bool IsApplicable(string hostname)
     {
-      return false;
+      return this.FindCredHelperName(hostname) != null;
     }
 
     /// <inheritdoc />
     public IDockerRegistryAuthenticationConfiguration GetAuthConfig(string hostname)
     {
-      return null;
+      this.logger.SearchingDockerRegistryCredential("CredsHelper");
+      var credHelperName = this.FindCredHelperName(hostname);
```

Comment:
I think we should consider the order here too. It's probably similar to the `Base64Provider` implementation (L:57-61).  If I'm not wrong, the last item should be used.